nucypher/policy/maps.py

```diff
@@ -103,329 +205,112 @@ def from_bytes(cls, data: bytes):
-                                 bob_verifying_key=bob.public_keys(SigningPower),
-                                 label=label)
 
-        # Encrypt each kfrag for an Ursula.
-        for ursula, verified_kfrag in zip(ursulas, verified_kfrags):
-            treasure_map.add_kfrag(ursula=ursula,
-                                   verified_kfrag=verified_kfrag,
-                                   publisher_stamp=publisher.stamp)
+        return TreasureMap.from_bytes(map_in_the_clear)
 
-        # Sign the map if needed before sending it out into the world.
-        treasure_map.prepare_for_publication(bob_encrypting_key=bob.public_keys(DecryptingPower),
-                                             publisher_stamp=publisher.stamp)
-
-        return treasure_map
-
-
-# FIXME: a dirty hack to make the tests pass. Fix it ASAP.
-# The problem with __new__ is that it does not get called before the first object of the class
-# is instantiated, so when we call `from_bytes()` for the first time, the `_splitters` dict
-# needs to already be populated.
-TreasureMap._splitters = {
-    'unversioned': BytestringKwargifier(TreasureMap,
-                                        public_signature=signature_splitter,
-                                        hrac=(bytes, HRAC_LENGTH),
-                                        message_kit=(UmbralMessageKit, VariableLengthBytestring),
-                                        ),
-    1: TreasureMapSplitter(TreasureMap,
-                           public_signature=signature_splitter,
-                           hrac=(bytes, HRAC_LENGTH),
-                           message_kit=(UmbralMessageKit, VariableLengthBytestring),
-                           )
-    }
-
-
-class SignedTreasureMap(TreasureMap):
-
-    _BRAND = b'SM'
-
-    def __init__(self, blockchain_signature=NOT_SIGNED, *args, **kwargs):
-        self._blockchain_signature = blockchain_signature
-        super().__init__(*args, **kwargs)
-
-    def include_blockchain_signature(self, blockchain_signer):
-        if self._payload is None:
-            self._set_payload()
-        self._blockchain_signature = blockchain_signer(self._payload)
-
-    def verify_blockchain_signature(self, checksum_address):
-        self._set_payload()
-        return verify_eip_191(message=self._payload,
+    def __bytes__(self):
+        return (bytes(self._public_signature) +
+                bytes(self.hrac) +
+                bytes(VariableLengthBytestring(self._encrypted_tmap.to_bytes())) +
+                # Other variants:
+                # - store all zeros if no blockchain signature
+                # - store the flag too in addition to all zeros, for added sanity check
+                # - use a faux blockchain signer in federated case
+                (b'\x00' if self._blockchain_signature is None else (b'\x01' + bytes(self._blockchain_signature)))
+                )
+
+    def verify_blockchain_signature(self, checksum_address: ChecksumAddress) -> bool:
```

Comment:
Adding to the above comment, if an offchain EIP-191 signature can be implemented, this method can simply be `verify`.

Reply:
Filed #2778 . Depending on how it is handled, we can change the method names.